src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
```

Comment:
The `index` variable is increased in each iteration, regardless of whether the current element of `toFormat` array is a date part or a separator. This can lead to incorrect indexing of `dateArr` array. Consider increasing `index` only when the current element of `toFormat` array is a date part.